pandas/tests/io/parser/test_parse_dates.py

```diff
@@ -896,3 +901,30 @@ def test_parse_delimited_date_swap(all_parsers, datestring,
     result = parser.read_csv(StringIO(datestring), header=None,
                              dayfirst=dayfirst, parse_dates=[0])
     tm.assert_frame_equal(result, expected)
+
+
+gen_random_datetime = st.dates(
+    min_value=date(1000, 1, 1),
+    max_value=date(9999, 12, 31)
+)
+_DEFAULT_DATETIME = datetime(1, 1, 1).replace(hour=0, minute=0,
+                                              second=0, microsecond=0)
+
+
+@given(gen_random_datetime)
+@pytest.mark.parametrize("date_format, delimiters, dayfirst", [
+    ("%m %d %Y", " -./", False),
```

Comment:
If you're going the `pytest.mark.parametrize` route, you can parametrize like this:

```python
@pytest.mark.parametrize("delimiter", list(" -./"))
@pytest.mark.parametrize("dayfirst", [True, False])
@pytest.mark.parametrize("date_format", [
    "%m %d %Y",
    "%d %m %Y",
    "%m %Y"
])
```

I think it would also be a good idea to expand this to *other* formats not necessarily covered by this particular PR, such as "%y%m%d", "%y %m %d", "%Y%m%d".

One of the more common side-effects of changing some parsing logic is that some unrelated format gets interpreted as your special case and causes problems.

Reply:
Thanks for this way using pytest. Changed and added "%Y %m %d" format. These formats: "%y%m%d", "%y %m %d", "%Y%m%d" do not pass on length. So, these cases can be tested without hypothesis I think.

Reply:
Added all additional formats, that you say, into hypothesis test